frontend/src/api/k8s/templates.ts

```diff
@@ -39,18 +46,42 @@ export const assembleServingRuntimeTemplate = (
   };
 };
 
-export const listTemplates = async (
-  namespace?: string,
-  labelSelector?: string,
-): Promise<TemplateKind[]> => {
-  const queryOptions = {
-    ...(namespace && { ns: namespace }),
-    ...(labelSelector && { queryParams: { labelSelector } }),
-  };
-  return k8sListResource<TemplateKind>({
-    model: TemplateModel,
-    queryOptions,
-  }).then((listResource) => listResource.items);
+export const useTemplates = (namespace?: string): CustomWatchK8sResult<TemplateKind[]> => {
+  const modelServingEnabled = useModelServingEnabled();
+  const customServingRuntimesEnabled = useCustomServingRuntimesEnabled();
+
+  const initResource: WatchK8sResource | null = namespace
+    ? {
+        isList: true,
+        groupVersionKind: groupVersionKind(TemplateModel),
+        namespace,
+      }
+    : null;
```

Comment:
initResource should be null when `modelServingEnabled` is false?
```suggestion
  const initResource: WatchK8sResource | null = namespace && modelServingEnabled
    ? {
        isList: true,
        groupVersionKind: groupVersionKind(TemplateModel),
        namespace,
      }
    : null;
```